skimage/feature/haar.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import division
+
+import numpy as np
+
+from ..draw import rectangle
+from ..exposure import rescale_intensity
+from ..transform import integral_image, integrate
+from .._shared.utils import check_random_state
+
+
+def haar_like_feature_coord(feature_type, height, width):
+    """Compute the coordinates of Haar-like features.
+
+    Parameters
+    ----------
+    feature_type : string
+        The type of feature to consider:
+
+        - 'type-2-x': 2 rectangles varying along the x axis;
+        - 'type-2-y': 2 rectangles varying along the y axis;
+        - 'type-3-x': 3 rectangles varying along the x axis;
+        - 'type-3-y': 3 rectangles varying along the y axis;
+        - 'type-4': 4 rectangles varying along x and y axis.
+
+    height : int
+        Height of the detection window.
+
+    width : int
+        Width of the detection window.
+
+    Returns
+    -------
+    feature_coord : list of tuple coord, shape (n_rectangles, 2, n_features)
+        Coordinates of the rectangles for each feature.
+
+    """
+    if feature_type == 'type-2-x':
+        height_feature = range(1, height, 1)
+        width_feature = range(2, width, 2)
+        output = [[[], []], [[], []]]
+    elif feature_type == 'type-2-y':
+        height_feature = range(2, height, 2)
+        width_feature = range(1, width, 1)
+        output = [[[], []], [[], []]]
+    elif feature_type == 'type-3-x':
+        height_feature = range(1, height, 1)
+        width_feature = range(3, width, 3)
+        output = [[[], []], [[], []], [[], []]]
+    elif feature_type == 'type-3-y':
+        height_feature = range(3, height, 3)
+        width_feature = range(1, width, 1)
+        output = [[[], []], [[], []], [[], []]]
+    elif feature_type == 'type-4':
+        height_feature = range(2, height, 2)
+        width_feature = range(2, width, 2)
+        output = [[[], []], [[], []], [[], []], [[], []]]
+    else:
+        known_feature = ('type-2-x', 'type-2-y',
+                         'type-3-x', 'type-3-y',
+                         'type-4')
+        raise ValueError('The type of feature is unknown. Got {} instead of'
+                         ' one of {}'.format(feature_type, known_feature))
+
+    for y in range(height):
```

Comment:
The PR looks fine, except for this unpacked loop which gives me the heebie-jeebies.  Is there any way we can script this in a single, parameterized loop for all cases?

Reply:
I have to think if a meshgrid or other type of grid could be used for that matter

Reply:
I should clarify that with "single", I meant "an additional".